services/web/src/lib/reusable-components/dropdown.tsx

```diff
@@ -0,0 +1,59 @@
+import { Select, SelectContent, SelectGroup, SelectItem, SelectTrigger, SelectValue } from '@/components/ui/select';
+import { FormField, FormItem, FormLabel, FormMessage } from '@/components/ui/form';
+
+type dropdownInput = {
+    name: string;
+    value: string;
+};
+
+type DropdownComponentProps = {
+    label: string;
+    placeholder: string;
+    className: string;
+    items: dropdownInput[];
+};
+
+export function DropdownComponent({ label, placeholder, className, items }: DropdownComponentProps) {
+    return (
+        <FormField
+            name="fruit"
+            render={({ field }) => (
+                <FormItem>
+                    <FormLabel>{label}</FormLabel>
+                    <Select onValueChange={field.onChange} defaultValue={field.value}>
+                        <SelectTrigger className={className}>
+                            <SelectValue placeholder={placeholder} />
+                        </SelectTrigger>
+                        <SelectContent>
+                            <SelectGroup>
+                                {items.map((item: dropdownInput) => (
```

Comment:
The type of item should automatically be inferred. If not, something is wrong with your code.